Fix exception handling for Hive OPTIMIZE table procedure

brandboat opened the pull request:
This is a fix after https://github.com/trinodb/trino/pull/9665/commits/57e48d2e78f35da01b6e1bd5a657dc79c2672ce5

Currently `someDeleted` flag is always false.

losipiuk commented:
LGTM thank you. Can you change the commit message to "Fix exception handling for Hive OPTIMIZE table procedure" ?
The flag was actually never set correctly.

brandboat commented:
> LGTM thank you. Can you change the commit message to "Fix exception handling for Hive OPTIMIZE table procedure" ?

I've addressed your comment in the latest commit, many thanks !

> The flag was actually never set correctly.

Actually I have another question, do we need to handle `fs.delete` when it return false ?

losipiuk commented:
> Actually I have another question, do we need to handle `fs.delete` when it return false ?

Technically no I think. But we expect the files which we are deleting to be in place. If they are not, it would mean someone deleted those behind the scenes as query was running. Not an expected usecase so I do not believe we must care about it.
